src/chi-vue/src/components/data-table/DataTable.tsx

```diff
@@ -0,0 +1,803 @@
+            <div class={SR_ONLY}>Select row {rowId}</div>
+          </label>
+        </div>
+      </div>
+    );
+  }
+
+  toggleRow(id: string) {
+    if (this.accordionsShown.includes(id)) {
+      this.accordionsShown.splice(this.accordionsShown.indexOf(id), 1);
+    } else {
+      this.accordionsShown.push(id);
+    }
+  }
+
+  rowAccordionContent(accordionData: any) {
+    if (accordionData.template) {
+      const template = this.$scopedSlots[accordionData.template]!(
+        accordionData.payload
+      );
+
+      return (
+        <div class={`${DATA_TABLE_CLASSES.ROW_CHILD} -p--2`} role="row">
+          {template}
+        </div>
+      );
+    } else if (accordionData.table) {
+      if (accordionData.table.data) {
+        return accordionData.table.data.map((bodyRow: DataTableRow) => {
+          return this.row(bodyRow, true);
+        });
+      }
+    } else {
+      return (
+        <div class={`${DATA_TABLE_CLASSES.ROW_CHILD} -p--2`} role="row">
+          {accordionData.value}
+        </div>
+      );
+    }
+  }
+
+  row(bodyRow: DataTableRow, rowChild = false) {
+    const row = [],
+      rowCells = [],
+      rowAccordionContent = [],
+      rowId = bodyRow.id ? `row-${bodyRow.id}` : `row-${uuid4()}`,
+      rowClass = rowChild
+        ? DATA_TABLE_CLASSES.ROW_CHILD
+        : DATA_TABLE_CLASSES.ROW;
+
+    if (this._expandable) {
+      if (bodyRow.accordion) {
+        rowCells.push(this.expansionButton(rowId));
+        rowAccordionContent.push(this.rowAccordionContent(bodyRow.accordion));
+      } else {
+        rowCells.push(
+          <div
+            class={`${DATA_TABLE_CLASSES.CELL} ${DATA_TABLE_CLASSES.EXPANDABLE}`}
+            role="cell"
+          />
+        );
+      }
+    }
+
+    if (this.data.config.selectable) {
+      rowCells.push(this.selectRowCheckbox(rowId, false));
+    }
+
+    let cellIndex = 0;
+
+    bodyRow.data.map((rowCell: any) => {
+      const columnSettings = this.data.head[
+          Object.keys(this.data.head)[cellIndex]
+        ],
+        alignment = this.cellAlignment(
+          rowCell.align ? rowCell.align : columnSettings.align || null
+        ),
+        cellLabel = rowCell.label || columnSettings.label,
+        cellWidth =
+          this.data.config.columnSizes && this.currentScreenBreakpoint
+            ? this.data.config.columnSizes[this.currentScreenBreakpoint][
+                cellIndex
+              ]
+            : null;
+      let cellData;
+
+      if (rowCell.template && this.$scopedSlots[rowCell.template]) {
+        if (rowCell.payload) {
+          cellData = this.$scopedSlots[rowCell.template]!(rowCell.payload);
+        }
+      } else if (typeof rowCell === 'string') {
+        cellData = (
+          <Tooltip message={rowCell} class="-w--100">
+            <div class={UTILITY_CLASSES.TYPOGRAPHY.TEXT_TRUNCATE}>
+              {rowCell}
+            </div>
+          </Tooltip>
+        );
+      } else {
```

Comment:
@janibolkvadze  This might fail if a rowCell value is a number, or null, or undefined. You should check that typeof rowCell is object and it has a `value` property.

Reply:
Last else case might print an empty cell, or an error cell.